projects/ui/src/state/beanstalk/sun/morning.ts

```diff
@@ -1,116 +1,122 @@
+ *
+ * ------------------------
+ *
+ * MorningFieldUpdater:
+ * - file: ~/state/beanstalk/field/morning.ts
+ * - function: <MorningFieldUpdater />
+ *
+ * We fetch the field at the start & end of the morning to ensure that the maxTemperature & totalSoil
+ * are updated. We fetch these values at the start and end of the morning.
+ *
+ * In addition, we also fetch & update the soil available every 4 seconds during the morning.
+ *
+ */
 
 export const BLOCKS_PER_MORNING = 25;
-
 export const FIRST_MORNING_BLOCK = 1;
-
 export const APPROX_SECS_PER_BLOCK = 12;
 
 export const getIsMorningInterval = (interval: BigNumber) =>
   interval.gte(FIRST_MORNING_BLOCK) && interval.lte(BLOCKS_PER_MORNING);
 
-export function useFetchMorningField() {
-  const beanstalk = useBeanstalkContract();
-  const dispatch = useDispatch();
-  const [fetchLatestBlock] = useFetchLatestBlock();
-
-  const fetch = useCallback(async () => {
-    try {
-      if (beanstalk) {
-        console.debug(
-          `[beanstalk/sun/useFetchMorningField] FETCH (contract = ${beanstalk.address})`
-        );
-        const [adjustedTemperature, maxTemperature, soil, blockData] =
-          await Promise.all([
-            beanstalk.temperature().then(tokenResult(BEAN)), // FIX ME
-            beanstalk.maxTemperature().then(tokenResult(BEAN)), // FIX ME
-            beanstalk.totalSoil().then(tokenResult(BEAN)),
-            fetchLatestBlock(),
-            // beanstalk
-          ]);
-
-        console.debug('[beanstalksun/useFetchMorningField] RESULT = ', {
-          scaledTemperature: adjustedTemperature.toString(),
-          maxTemperature: maxTemperature.toString(),
-          soil: soil.toString(),
-        });
-
-        dispatch(updateTotalSoil(soil));
-        dispatch(updateScaledTemperature(adjustedTemperature));
-        dispatch(updateMaxTemperature(maxTemperature));
-        dispatch(updateMorningBlock(blockData.blockNumber));
-
-        return [adjustedTemperature, maxTemperature, soil, blockData] as const;
-      }
-      return [undefined, undefined, undefined, undefined] as const;
-    } catch (e) {
-      console.debug('[beanstalk/sun/useFetchMorningField] FAILED', e);
-      console.error(e);
-      return [undefined, undefined, undefined, undefined] as const;
-    }
-  }, [beanstalk, dispatch, fetchLatestBlock]);
-
-  return [fetch] as const;
-}
+function useUpdateMorning() {
+  const morningTime = useAppSelector((s) => s._beanstalk.sun.morningTime);
+  const season = useAppSelector((s) => s._beanstalk.sun.season);
+  const morning = useAppSelector((s) => s._beanstalk.sun.morning);
 
-export default function MorningUpdater() {
-  const morningTime = useSelector(selectMorningBlockTime);
-  const blockMap = useSelector(selectMorningBlockMap);
-  const morning = useSelector(selectMorning);
+  const [_, { calculate: calculateTemperature }] = useTemperature();
+  const [_soilData, { calculate: calculateNextSoil }] = useSoil();
```

Comment:
Curious why the temperature is calculated for the current block, but soil is calculated for the next block during the morning

Reply:
so we calculate the temperature for the next block via 
`const scaledTemp = calculateTemperature(morningBlock.plus(1))`

I chose to calculate for the next morningBlock b/c the calculation requires knowledge of the temperature of the current block & the temperature of the next block. calculating 'soil' instead of 'nextSoil' would cause issues If deltaBlocks = 0, calculating soil would require, whereas if deltaBlocks = 25, we can always assume nextTemperature is the maxTemperature.